From 3e4961cdb328fb2c9ba500a17192da8ca6fe8eee Mon Sep 17 00:00:00 2001
From: Rui Miguel Silva <rui.silva@linaro.org>
Date: Fri, 29 Jun 2018 15:18:01 +0100
Subject: [PATCH 03/24] optee: adjust dependencies and default values for dram

We may have, the not yet considered, scenario where OPTEE is loaded before
u-boot and *not* by u-boot, e.g, the boot flow using the ARM Trusted
Firmware (ATF), where in the 32bit flow is:
BootRom->ATF(BL2)->Optee(BL32)->u-boot(BL33)

In this case we need still to reserve the memory used by optee, to avoid
for example to realocate ourself to the same address at the end of DRAM.
So, we change here the dependencies on the OPTEE lib and we set the default
size and base of TZRAM to zero.

Signed-off-by: Rui Miguel Silva <rui.silva@linaro.org>
Signed-off-by: Bryan O'Donoghue <bryan.odonoghue@linaro.org>
Cc: Fabio Estevam <fabio.estevam@nxp.com>
Cc: Ryan Harkin <ryan.harkin@linaro.org>
Cc: u-boot@lists.denx.de
---
 configs/warp7_defconfig | 2 ++
 lib/optee/Kconfig       | 8 ++++----
 2 files changed, 6 insertions(+), 4 deletions(-)

diff --git a/configs/warp7_defconfig b/configs/warp7_defconfig
index 15a66739f8..a1c0b69ea8 100644
--- a/configs/warp7_defconfig
+++ b/configs/warp7_defconfig
@@ -46,5 +46,7 @@ CONFIG_USB_ETH_CDC=y
 CONFIG_USBNET_HOST_ADDR="de:ad:be:af:00:00"
 CONFIG_OF_LIBFDT=y
 CONFIG_OPTEE=y
+CONFIG_OPTEE_TZDRAM_SIZE=0x3000000
+CONFIG_OPTEE_TZDRAM_BASE=0x9d000000
 CONFIG_OPTEE_LOAD_ADDR=0x84000000
 CONFIG_BOOTM_OPTEE=y
diff --git a/lib/optee/Kconfig b/lib/optee/Kconfig
index 1e5ab45c3d..3773d89c31 100644
--- a/lib/optee/Kconfig
+++ b/lib/optee/Kconfig
@@ -10,21 +10,20 @@ config OPTEE
 config OPTEE_LOAD_ADDR
 	hex "OPTEE load address"
 	default 0x00000000
+	depends on OPTEE
 	help
 	  The load address of the bootable OPTEE binary.
 
 config OPTEE_TZDRAM_SIZE
 	hex "Amount of Trust-Zone RAM for the OPTEE image"
-	depends on OPTEE
-	default 0x3000000
+	default 0x0000000
 	help
 	  The size of pre-allocated Trust Zone DRAM to allocate for the OPTEE
 	  runtime.
 
 config OPTEE_TZDRAM_BASE
 	hex "Base address of Trust-Zone RAM for the OPTEE image"
-	depends on OPTEE
-	default 0x9d000000
+	default 0x00000000
 	help
 	  The base address of pre-allocated Trust Zone DRAM for
 	  the OPTEE runtime.
@@ -32,6 +31,7 @@ config OPTEE_TZDRAM_BASE
 config BOOTM_OPTEE
 	bool "Support OPTEE bootm command"
 	select BOOTM_LINUX
+	depends on OPTEE
 	default n
 	help
 	  Select this command to enable chain-loading of a Linux kernel
-- 
2.23.0.rc1

